### ISSUE TITLE
command to update cached git repos

### ISSUE THREAD

**Issue by matthiaskrgr** [matthiaskrgr/cargo-cache#18]

**Comment by matthiaskrgr** [comment]
Note: I tried implementing this (on top of b69f1f53a49b0a379fb48aea699dcef17950e9b2 )

````diff
diff --git a/src/library.rs b/src/library.rs
index ca221a8..731dea3 100644
--- a/src/library.rs
+++ b/src/library.rs
@@ -322,6 +322,33 @@ pub(crate) fn pad_strings(
     formatted_line
 }

+pub(crate) fn update_repos(ccp: &CargoCachePaths) {
+    println!("updating repos!");
+    let git_db = &ccp.git_repos_bare;
+    println!("{:?}", git_db);
+    for repo in std::fs::read_dir(git_db).unwrap() {
+        let repo_path = repo.unwrap().path();
+        println!("updating repo: {}", repo_path.display());
+        let repo = match git2::Repository::open_bare(&repo_path) {
+            Ok(repo) => repo,
+            Err(e) => {
+                eprintln!(
+                    "Error opening repo at {}: \"{}\", skipping.",
+                    repo_path.display(),
+                    e
+                );
+                continue;
+            } // update all repos here
+        };
+        //        let remotes = repo.remotes();
+        println!(
+            "remote {:?}",
+            repo.find_remote("matthiaskrgr").unwrap().name()
+        );
+    }
+
+}
+
 #[cfg(test)]
 mod libtests {
     use super::*;
diff --git a/src/main.rs b/src/main.rs
index f124704..497c9e1 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -82,6 +82,9 @@ fn main() {
         }
     };

+    library::update_repos(&cargo_cache);
+    std::process::exit(1);
+
     if config.is_present("list-dirs") {
         // only print the directories and exit, don't calculate anything else
         println!("{}", cargo_cache);
````
however, to call the "git fetch" of cargo 
https://github.com/rust-lang/cargo/blob/2b41dd4/src/cargo/sources/git/utils.rs#L670
I would need to have the repo url as parameter which is not stored anywhere inside the repo because it is always taken from cargos dependency information.

tl;dr cannot update repo without having a specific build depencency
